lib/pbench/test/unit/server/query_apis/test_datasets.py

```diff
@@ -0,0 +1,204 @@
+    constructor and `post` service.
+    """
+
+    @pytest.fixture(autouse=True)
+    def _setup(self, client):
+        super()._setup(
+            cls_obj=Datasets(client.config),
+            pbench_endpoint="/datasets/random_md5_string1",
+            elastic_endpoint="/_delete_by_query?ignore_unavailable=true&refresh=true",
+            index_from_metadata=None,
+            empty_es_response_payload=self.EMPTY_ES_RESPONSE_PAYLOAD,
+            api_method=ApiMethod.DELETE,
+            use_index_map=True,
+        )
+
+    @pytest.mark.parametrize(
+        "user, expected_status",
+        [
+            ("drb", HTTPStatus.OK),
+            ("test_admin", HTTPStatus.OK),
+            ("test", HTTPStatus.FORBIDDEN),
+            (None, HTTPStatus.UNAUTHORIZED),
+        ],
+    )
+    def test_empty_delete(
+        self,
+        client,
+        server_config,
+        query_api,
+        find_template,
+        user,
+        expected_status,
+        get_token_func,
+    ):
+        """Check deletion with no Elasticsearch documents"""
+
+        headers = None
+
+        if user:
+            token = get_token_func(user)
+            assert token
+            headers = {"authorization": f"bearer {token}"}
+
+        index = self.build_index_from_metadata()
+
+        response = query_api(
+            self.pbench_endpoint,
+            self.elastic_endpoint,
+            payload=None,
+            expected_index=index,
+            expected_status=expected_status,
+            request_method=self.api_method,
+            json=EMPTY_DELDATE_RESPONSE,
+            headers=headers,
+        )
+        assert response.status_code == expected_status
+        if response.status_code == HTTPStatus.OK:
+            res_json = response.json
+            expected = {
+                "total": 0,
+                "updated": 0,
+                "deleted": 0,
+                "failures": 0,
+                "version_conflicts": 0,
+            }
+            assert expected == res_json
+
+            # On success, the dataset should be gone
+            with pytest.raises(DatasetNotFound):
+                Dataset.query(name="drb")
+        else:
+            # On failure, the dataset should still exist
+            assert Dataset.query(name="drb")
+            assert response.json["message"].endswith(
+                "is not authorized to DELETE a resource owned by drb with private access"
+            )
+
+    @pytest.mark.parametrize(
+        "user, expected_status",
+        [
+            ("drb", HTTPStatus.OK),
+            ("test_admin", HTTPStatus.OK),
+            ("test", HTTPStatus.FORBIDDEN),
+            (None, HTTPStatus.UNAUTHORIZED),
+        ],
+    )
+    def test_empty_update_access(
+        self,
+        client,
+        server_config,
+        query_api,
+        find_template,
+        user,
+        expected_status,
+        get_token_func,
+    ):
+        """Check deletion with no Elasticsearch documents"""
```

Comment:
I presume the docstring should say "update" rather than "deletion"....

Reply:
You realize you're not supposed to be reviewing yet, right? 😆